timemachine/parallel/client.py

```diff
@@ -238,3 +239,94 @@ def verify(self):
                     assert False, f"Host {host} '{field}' new value of {new_val} != {prev_vals[field]}"
                 else:
                     prev_vals[field] = new_val
+
+
+class AbstractFileClient:
+    def store(self, path: str, data: bytes) -> str:
+        """
+        Store the results to the given path.
+
+        Parameters
+        ----------
+        path:
+            Relative path to store the data. The client may interpret
+            this path as appropriate (i.e. file path, s3 path, etc).
+
+        data:
+            Binary contents to store.
+
+        Returns
+        -------
+        str
+            Full name of the result as stored.
+
+        """
+        raise NotImplementedError()
+
+    def load(self, path: str) -> bytes:
+        """
+        Load the results from the given path.
+
+        Parameters
+        ----------
+        path:
+            Path to load from, the value returned by the `store` method.
+
+        Returns
+        -------
+        bytes
+            Binary contents of the file.
+        """
+        raise NotImplementedError()
+
+    def exists(self, path: str) -> bool:
+        """
+        Parameters
+        ----------
+        path:
+            Path to load from, the value returned by the `store` method.
+
+        Returns
+        -------
+        bool
+            True if the results exist at this path.
+        """
+        raise NotImplementedError()
+
+
+class FileClient(AbstractFileClient):
+    def __init__(self, base: Path = None):
+        self.base = base or Path().cwd()
+
+    def store(self, path: str, data: bytes):
+        """
+        See abstract class for documentation.
+        """
+        full_path = Path(self.full_path(path))
+        full_path.write_bytes(data)
```

Comment:
The base class defines this as returning a string, but seems less useful to return. Should unify the behavior one way or another.

Reply:
removed str return, this was from an earlier version